[CI] FiltersAggregatorTests testTermOnFilterWithMatchAll failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/sqwgxceult4se/tests/:server:test/org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests/testTermOnFilterWithMatchAll

**Reproduction line:**
```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testTermOnFilterWithMatchAll" -Dtests.seed=BF147139171E081E -Dtests.locale=de-DE -Dtests.timezone=CST6CDT -Druntime.java=20
```

**Applicable branches:**
main

**Reproduces locally?:**
Yes

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests&tests.test=testTermOnFilterWithMatchAll

**Failure excerpt:**
```
java.lang.AssertionError: Expected a map containing
                       test: a map containing
             segments_counted: <1>
           segments_collected: <0>
segments_with_doc_count_field: <0>
   segments_with_deleted_docs: expected <0> but was <1>
                      filters: a list containing
                              0: a map containing
                            query: "foo:bar"
segments_counted_in_constant_time: <1>

  at __randomizedtesting.SeedInfo.seed([BF147139171E081E:3DC379DE6B047556]:0)
  at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:82)
  at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:65)
  at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.lambda$testTermOnFilterWithMatchAll$32(FiltersAggregatorTests.java:824)
  at org.elasticsearch.search.aggregations.AggregatorTestCase.debugTestCase(AggregatorTestCase.java:846)
  at org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testTermOnFilterWithMatchAll(FiltersAggregatorTests.java:817)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:578)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1623)

```

--- a/server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java
@@ -64,7 +64,7 @@ public class AggregationCollectorTests extends AggregatorTestCase {
             builder,
             new MatchAllDocsQuery(),
             iw -> {},
-            (indexSearcher, agg) -> assertEquals(expected, agg.scoreMode().needsScores())
+            (indexReader, agg) -> assertEquals(expected, agg.scoreMode().needsScores())
         );
     }
 

--- a/server/src/test/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregatorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/bucket/filter/FiltersAggregatorTests.java
@@ -20,7 +20,6 @@ import org.apache.lucene.index.Term;
 import org.apache.lucene.search.BooleanClause.Occur;
 import org.apache.lucene.search.BooleanQuery;
 import org.apache.lucene.search.IndexOrDocValuesQuery;
-import org.apache.lucene.search.IndexSearcher;
 import org.apache.lucene.search.MatchAllDocsQuery;
 import org.apache.lucene.search.MatchNoDocsQuery;
 import org.apache.lucene.search.Query;
@@ -134,7 +133,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
         if (askForOtherBucket) {
             builder.otherBucket(true).otherBucketKey("other");
         }
-        withAggregator(builder, new MatchAllDocsQuery(), iw -> {}, (searcher, aggregator) -> {
+        withAggregator(builder, new MatchAllDocsQuery(), iw -> {}, (reader, aggregator) -> {
             InternalFilters result = (InternalFilters) aggregator.buildEmptyAggregation();
             for (int i = 0; i < filters.length; i++) {
                 assertThat(result.getBucketByKey(String.valueOf(i)).getDocCount(), equalTo(0L));
@@ -308,7 +307,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
             iw.addDocument(List.of(new LongPoint("test", time), new SortedNumericDocValuesField("test", time)));
             time = DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.parseMillis("2020-02-01");
             iw.addDocument(List.of(new LongPoint("test", time), new SortedNumericDocValuesField("test", time)));
-        }, (searcher, aggregator) -> {
+        }, (reader, aggregator) -> {
             /*
              * The topLevelQuery is entirely contained within the filter query so
              * it is good enough to match that. See MergedPointRangeQueryTests for
@@ -661,9 +660,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                     bitsetFilterCache,
                     LongPoint.newRangeQuery("t", 5, Long.MAX_VALUE)
                 );
-                IndexSearcher searcher = newSearcher(limitedReader);
-                int segmentsWithLiveDocs = (int) searcher.getIndexReader()
-                    .leaves()
+                int segmentsWithLiveDocs = (int) limitedReader.leaves()
                     .stream()
                     .map(LeafReaderContext::reader)
                     .filter(leafReader -> leafReader.getLiveDocs() != null)
@@ -687,7 +684,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                                         "filters",
                                         matchesList().item(
                                             matchesMap().entry("query", "*:*")
-                                                .entry("segments_counted_in_constant_time", searcher.getLeafContexts().size())
+                                                .entry("segments_counted_in_constant_time", limitedReader.leaves().size())
                                         )
                                     )
                             )
@@ -735,9 +732,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                     bitsetFilterCache,
                     LongPoint.newRangeQuery("t", 5, Long.MAX_VALUE)
                 );
-                IndexSearcher searcher = newSearcher(limitedReader);
-                int segmentsWithLiveDocs = (int) searcher.getIndexReader()
-                    .leaves()
+                int segmentsWithLiveDocs = (int) limitedReader.leaves()
                     .stream()
                     .map(LeafReaderContext::reader)
                     .filter(leafReader -> leafReader.getLiveDocs() != null)
@@ -761,7 +756,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                                         "filters",
                                         matchesList().item(
                                             matchesMap().entry("query", "foo:bar")
-                                                .entry("segments_counted_in_constant_time", lessThan(searcher.getLeafContexts().size()))
+                                                .entry("segments_counted_in_constant_time", lessThan(limitedReader.leaves().size()))
                                         )
                                     )
                             )
@@ -806,9 +801,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                     bitsetFilterCache,
                     LongPoint.newRangeQuery("t", Long.MIN_VALUE, Long.MAX_VALUE)
                 );
-                IndexSearcher searcher = newSearcher(limitedReader);
-                int segmentsWithLiveDocs = (int) searcher.getIndexReader()
-                    .leaves()
+                int segmentsWithLiveDocs = (int) limitedReader.leaves()
                     .stream()
                     .map(LeafReaderContext::reader)
                     .filter(leafReader -> leafReader.getLiveDocs() != null)
@@ -833,7 +826,7 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
                                         "filters",
                                         matchesList().item(
                                             matchesMap().entry("query", "foo:bar")
-                                                .entry("segments_counted_in_constant_time", searcher.getLeafContexts().size())
+                                                .entry("segments_counted_in_constant_time", limitedReader.leaves().size())
                                         )
                                     )
                             )
@@ -1599,10 +1592,10 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
         };
         // Exists queries convert to MatchNone if this isn't defined
         FieldNamesFieldMapper.FieldNamesFieldType fnft = FieldNamesFieldMapper.FieldNamesFieldType.get(true);
-        withAggregator(builder, new MatchAllDocsQuery(), buildIndex, (searcher, aggregator) -> {
+        withAggregator(builder, new MatchAllDocsQuery(), buildIndex, (reader, aggregator) -> {
             assertThat(aggregator, instanceOf(FilterByFilterAggregator.class));
 
-            Map<String, Object> debug = collectAndGetFilterDebugInfo(searcher, aggregator);
+            Map<String, Object> debug = collectAndGetFilterDebugInfo(reader, aggregator);
             assertMap(debug, matchesMap().extraOk().entry("segments_counted_in_constant_time", greaterThan(0)));
         }, fieldType, fnft);
         testCase(buildIndex, (InternalFilters result) -> {
@@ -1616,9 +1609,9 @@ public class FiltersAggregatorTests extends AggregatorTestCase {
         return MOCK_OBJECT_MAPPERS;
     }
 
-    private Map<String, Object> collectAndGetFilterDebugInfo(IndexSearcher searcher, Aggregator aggregator) throws IOException {
+    private Map<String, Object> collectAndGetFilterDebugInfo(IndexReader reader, Aggregator aggregator) throws IOException {
         aggregator.preCollection();
-        for (LeafReaderContext ctx : searcher.getIndexReader().leaves()) {
+        for (LeafReaderContext ctx : reader.leaves()) {
             LeafBucketCollector leafCollector = aggregator.getLeafCollector(new AggregationExecutionContext(ctx, null, null, null));
             assertTrue(leafCollector.isNoop());
         }

--- a/server/src/test/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregatorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/bucket/histogram/DateHistogramAggregatorTests.java
@@ -1118,7 +1118,7 @@ public class DateHistogramAggregatorTests extends DateHistogramAggregatorTestCas
             new DateHistogramAggregationBuilder("test").field(AGGREGABLE_DATE).calendarInterval(DateHistogramInterval.YEAR).offset(10),
             new MatchAllDocsQuery(),
             iw -> {},
-            (searcher, aggregator) -> {
+            (reader, aggregator) -> {
                 InternalDateHistogram histo = (InternalDateHistogram) aggregator.buildEmptyAggregation();
                 /*
                  * There was a time where we including the offset in the

--- a/server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java
+++ b/server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/TermsAggregatorTests.java
@@ -1891,7 +1891,7 @@ public class TermsAggregatorTests extends AggregatorTestCase {
             terms1 = dh.getBuckets().get(1).getAggregations().get("k");
             assertThat(terms1.getBuckets().stream().map(StringTerms.Bucket::getKey).collect(toList()), equalTo(List.of("a")));
         }, new AggTestConfig(builder, dft, kft));
-        withAggregator(builder, new MatchAllDocsQuery(), buildIndex, (searcher, aggregator) -> {
+        withAggregator(builder, new MatchAllDocsQuery(), buildIndex, (reader, aggregator) -> {
             TermsAggregator terms = (TermsAggregator) aggregator.subAggregator("k");
             Map<String, Object> info = new HashMap<>();
             terms.collectDebugInfo(info::put);

--- a/test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java
+++ b/test/framework/src/main/java/org/elasticsearch/search/aggregations/AggregatorTestCase.java
@@ -866,7 +866,7 @@ public abstract class AggregatorTestCase extends ESTestCase {
         AggregationBuilder aggregationBuilder,
         Query query,
         CheckedConsumer<RandomIndexWriter, IOException> buildIndex,
-        CheckedBiConsumer<IndexSearcher, Aggregator, IOException> verify,
+        CheckedBiConsumer<IndexReader, Aggregator, IOException> verify,
         MappedFieldType... fieldTypes
     ) throws IOException {
         try (Directory directory = newDirectory()) {
@@ -878,9 +878,8 @@ public abstract class AggregatorTestCase extends ESTestCase {
                 DirectoryReader unwrapped = DirectoryReader.open(directory);
                 DirectoryReader indexReader = wrapDirectoryReader(unwrapped)
             ) {
-                IndexSearcher searcher = newIndexSearcher(indexReader);
-                try (AggregationContext context = createAggregationContext(searcher, query, fieldTypes)) {
-                    verify.accept(searcher, createAggregator(aggregationBuilder, context));
+                try (AggregationContext context = createAggregationContext(indexReader, query, fieldTypes)) {
+                    verify.accept(indexReader, createAggregator(aggregationBuilder, context));
                 }
             }
         }

